ecs-agent/tmds/handlers/fis/v1/handlers/handlers.go

```diff
@@ -0,0 +1,148 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/metrics"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/tmds/handlers/fis/v1/types"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/tmds/handlers/utils"
+	v4 "github.com/aws/amazon-ecs-agent/ecs-agent/tmds/handlers/v4"
+	state "github.com/aws/amazon-ecs-agent/ecs-agent/tmds/handlers/v4/state"
+
+	"github.com/gorilla/mux"
+)
+
+type handler struct {
+	// TODO: Mutex will be used in a future PR
+	// mu             sync.Mutex
+	agentState     state.AgentState
+	metricsFactory metrics.EntryFactory
+}
+
+// RegisterFISHandlers will initialize all of the Start/Stop/Check fault injection endpoints and register them to the TMDS router
+func RegisterFISHandlers(muxRouter *mux.Router, agentState state.AgentState, metricsFactory metrics.EntryFactory) {
+	handler := handler{
+		agentState:     agentState,
+		metricsFactory: metricsFactory,
+	}
+
+	// Setting up handler endpoints for network blackhole port fault injections
+	muxRouter.HandleFunc(
+		FISNetworkFaultPath(types.BlackHolePortFaultType),
+		handler.StartNetworkBlackholePortHandler(),
```

Comment:
nit: `handler.StartNetworkBlackholePort()` should be good IMO. The second Handler looks not necessary.